src/mapwindow.jl

```diff
@@ -68,12 +68,19 @@ end
 
 mapwindow(f, img, window::AbstractArray, args...; kwargs...) = mapwindow(f, img, (window...,), args...; kwargs...)
 
+
+
 function mapwindow{T,N}(f,
                         img::AbstractArray{T,N},
                         window::Indices{N},
                         border::BorderSpecAny;
                         callmode=:copy!)
-    _mapwindow(replace_function(f), img, window, border, default_shape(f); callmode=callmode)
+    if(is_medianfilter(f))
```

Comment:
Instead of introducing `is_medianfilter`, a more "Julian" approach would be to change the signature of your `median_filter` function below to be the following method:
```julia
function mapwindow{T,N}(f::typeof(median!),
```
But I actually have a slightly different suggestion, see below.